### ISSUE TITLE
Train the model with my own data

### ISSUE THREAD

**Issue by ficht74** [lukas-blecher/LaTeX-OCR#281]
Hi, First of all thanks for the great work!
I want to train your model with my own data, but it doesn't work and I don't know why?
The training with your data works perfect and I have modified my data like yours Crohme and pdf.
The generation of the dataset works, also without problems, but if I start the training I get an error which tells me the tensors have different sizes? Please, do you have any tipps for me to solve the problem.
<img width="1200" alt="Bildschirmfoto 2023-06-12 um 13 08 07" src="https://github.com/lukas-blecher/LaTeX-OCR/assets/93684165/95b3fb62-9706-4349-93a5-d1cd5018a643">

**Comment by lukas-blecher** [comment]
Hi, thanks :)
You'll need to pad the images.
See #101

**Comment by ficht74** [comment]
That's it - mega thanx Lukas!!!
But I have an additional question regarding image dimension. My image height is often smaller than 32:
{'0000054.png': (96, 22),
 '0000599.png': (106, 36),
 '0000226.png': (94, 34),
 '0000058.png': (140, 19),
 '0000536.png': (92, 34),
 '0000087.png': (60, 19),
 '0000459.png': (142, 18),
 '0000608.png': (95, 21),
 '0000461.png': (44, 31),
 '0000438.png': (92, 18),
 '0000474.png': (79, 16),
 '0000311.png': (70, 18),
 '0000098.png': (68, 32),
 '0000535.png': (148, 37),
 '0000231.png': (104, 32),
 '0000353.png': (81, 19),
 '0000390.png': (129, 19)

Does it make sense to resize the images with a factor to reach at least a height of 32 which is the min_height? The training runs currently with the original dimension and It seems to work.

**Comment by lukas-blecher** [comment]
If you added the `pad` option, the images will be padded with white pixels up to the next number fully divisible by 32.
Resizing might introduce unwanted artifacts and skew the data distribution unnecessarily. I think padding will work fine.
Hope I didn't miss the point of your question.

**Comment by ficht74** [comment]
Hi Lukas, thanks a lot for your help. 

Can I ask you two more questions regarding training?

1.) The Training runs fine, but the acc and bleu scores are very low. Do you have an idea what I am doing wrong?
<img width="1136" alt="Bildschirmfoto 2023-06-12 um 22 18 59" src="https://github.com/lukas-blecher/LaTeX-OCR/assets/93684165/090c15a9-c19d-494a-aa59-2d4a8e407b4d">

2.) How can I load the best model after training and make predictions on my unseen test data?

Cheers, Christian

**Comment by lukas-blecher** [comment]
1) My dataset format is pretty stupid. I was just too lazy to change it after it worked once. Try to check for a couple samples if the text matches the image I guess.
2) `python -m pix2tex.eval --checkpoint LaTeX-OCR/checkpoints/LaTeX-OCR-tuned-cf/xx_step1999.pth --data path/to/test.pkl`
also if you're using a different architecture add `--config path/to/config.yaml`

**Comment by ficht74** [comment]
I tried it this way, as you mentioned in the previous post.

!python -m pix2tex.eval --checkpoint /content/drive/MyDrive/models/LaTeX-OCR/my_model_path/LaTeX-OCR-tuned-cf-allnew/LaTeX-OCR-tuned-cf-allnew_e10_step4144.pth --config /content/drive/MyDrive/models/LaTeX-OCR/my_model_path/LaTeX-OCR-tuned-cf-allnew/config.yaml --data /content/drive/MyDrive/models/LaTeX-OCR/data/val.pkl

But I get this error:
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/pix2tex/eval.py", line 1, in <module>
    from pix2tex.dataset.dataset import Im2LatexDataset
  File "/usr/local/lib/python3.10/dist-packages/pix2tex/dataset/dataset.py", line 1, in <module>
    import torch
  File "/usr/local/lib/python3.10/dist-packages/torch/__init__.py", line 933, in <module>
    from ._tensor import Tensor
  File "/usr/local/lib/python3.10/dist-packages/torch/_tensor.py", line 12, in <module>
    import torch.utils.hooks as hooks
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/__init__.py", line 5, in <module>
    from ._crash_handler import enable_minidumps, disable_minidumps, enable_minidumps_on_exceptions
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_crash_handler.py", line 3, in <module>
    import pathlib
  File "/usr/local/lib/python3.10/dist-packages/pathlib.py", line 10, in <module>
    from collections import Sequence
ImportError: cannot import name 'Sequence' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

basically, I don't want to use your eval method because I have my own. I only want to use pix2tex tuned version for predictions, because I have my own metrics functions for evaluation.

Do you have an idea where the problem could be?

**Comment by ficht74** [comment]
Hi Lukas, 
I think I found a solution for me now, hopefully it helps others too!
I am using the arguments of the LaTeXOCR Class, which are None as default setting and create new arguments for the tuned model weights and config.

from types import SimpleNamespace
from munch import Munch

arguments = SimpleNamespace(config=my_path/config.yaml', checkpoint='my_path/LaTeX-OCR-tuned.pth', no_cuda=True, no_resize=False)

my_model = pix2tex.LatexOCR(arguments)
my_model.args  -> let you can check if the correct model is loaded

Please could you check if I am right or if there is an easier or smarter way?

Thanx Christian

**Comment by lukas-blecher** [comment]
yes, that's how you load it in python
